### PR TITLE
feat(kuma-cp): remove value of secret when logging Secret Resources

### PR DESCRIPTION
**Pull request by Automaat**
Fixes: #2058

Signed-off-by: Marcin Skalski <marcin.skalski@konghq.com>

### Checklist prior to review

- [x] Link to docs PR or issue --
- [x] Link to UI issue or PR --
- [x] Is the [issue worked on linked][1]? --
- [x] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [x] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Unit Tests --
- [x] E2E Tests --
- [x] Manual Universal Tests --
- [x] Manual Kubernetes Tests --
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [ ] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [x] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

**Review by michaelbeaumont — APPROVED** [review]
LGTM

**Comment by lahabana** [comment]
LGTM I guess you've checked this manually?

**Review by lukidzi — APPROVED** [review]

File: pkg/core/resources/apis/mesh/external_service_helpers.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+	mesh_proto "github.com/kumahq/kuma/api/mesh/v1alpha1"
+)
+
+func (es *ExternalServiceResource) IsReachableFromZone(zone string) bool {
+	return es.Spec.Tags[mesh_proto.ZoneTag] == "" || es.Spec.Tags[mesh_proto.ZoneTag] == zone
+}
+
+func (esl *ExternalServiceResourceList) MarshalLog() interface{} {
+	maskedList := make([]*ExternalServiceResource, len(esl.Items))
```
[the remark this answers is not in the document]
Comment (lahabana) [reply pkg/core/resources/apis/mesh/external_service_helpers.go:14]: Or do `for i, es := range esl.Items {maskedList[i] = es.MarshalLog().(*ExternalServiceResource)}`

(Make without the 0 creates a 0 initialized array

File: pkg/core/resources/apis/mesh/external_service_helpers.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+		maskedList = append(maskedList, es.MarshalLog().(*ExternalServiceResource))
+	}
+	return ExternalServiceResourceList{
+		Items:      maskedList,
+		Pagination: esl.Pagination,
+	}
+}
+
+func (es *ExternalServiceResource) MarshalLog() interface{} {
+	c, err := copystructure.Copy(es)
```
Comment (lahabana) [inline pkg/core/resources/apis/mesh/external_service_helpers.go:25]: I think you can use protobuf copy instead (at least on the spec

File: pkg/core/resources/apis/mesh/mesh_helpers.go [hunk 3]
```diff
@@ -135,3 +139,48 @@ func ParseDuration(durationStr string) (time.Duration, error) {
 	}
 	return dur, nil
 }
+
+func (ml *MeshResourceList) MarshalLog() interface{} {
+	maskedList := make([]*MeshResource, len(ml.Items))
```
Comment (lahabana) [inline pkg/core/resources/apis/mesh/mesh_helpers.go:144]: Same append issue

File: pkg/core/resources/apis/system/secret_resource_helper.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+	"github.com/kumahq/kuma/pkg/core/resources/model"
+)
+
+type secretResource struct {
+	Meta model.ResourceMeta
+	Spec string
+}
+
+func (l *SecretResourceList) MarshalLog() interface{} {
+	list := make([]interface{}, len(l.Items))
```
Comment (lahabana) [inline pkg/core/resources/apis/system/secret_resource_helper.go:13]: same here

File: pkg/core/resources/apis/mesh/external_service_helpers.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Automaat) [reply pkg/core/resources/apis/mesh/external_service_helpers.go]: i didn't know that, changed now

File: pkg/core/resources/apis/mesh/external_service_helpers.go (hunk 2 above)
Comment (Automaat) [reply pkg/core/resources/apis/mesh/external_service_helpers.go]: changed

**Review by lahabana — APPROVED** [review]

**Comment by lahabana** [comment]
@mergifyio backport release-2.0 release-1.9 release-1.8 release-1.7 release-1.6

**Pull request merged**